### ISSUE TITLE
INDArray transpose requires org.nd4j.shade.jackson.module.jaxb.JaxbAnnotationIntrospector

### ISSUE THREAD

**Issue by cinqs** [deeplearning4j/nd4j#2155]
#### Issue Description

Please describe your issue, along with:
- make a Vector transpose without problem
- 

```console
org.reflections.ReflectionsException: could not get type for name org.nd4j.shade.jackson.module.jaxb.JaxbAnnotationIntrospector
	at org.reflections.ReflectionUtils.forName(ReflectionUtils.java:390)
	at org.reflections.Reflections.expandSuperTypes(Reflections.java:381)
	at org.reflections.Reflections.<init>(Reflections.java:126)
	at org.nd4j.linalg.compression.BasicNDArrayCompressor.loadCompressors(BasicNDArrayCompressor.java:41)
	at org.nd4j.linalg.compression.BasicNDArrayCompressor.<init>(BasicNDArrayCompressor.java:30)
	at org.nd4j.linalg.compression.BasicNDArrayCompressor.<clinit>(BasicNDArrayCompressor.java:23)
	at org.nd4j.linalg.factory.Nd4j.getCompressor(Nd4j.java:6306)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.permute(BaseNDArray.java:4682)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.transposei(BaseNDArray.java:3690)
	at org.nd4j.linalg.api.ndarray.BaseNDArray.transpose(BaseNDArray.java:3678)
```

#### Version Information

Please indicate relevant versions, including, if relevant:

```xml
<dependency>
		  <groupId>org.nd4j</groupId>
		  <artifactId>nd4j-native-platform</artifactId>
		  <version>0.9.1</version>
</dependency>
<dependency>
                 <groupId>org.deeplearning4j</groupId>
                 <artifactId>deeplearning4j-core</artifactId>
                 <version>0.9.1</version>
</dependency>
```

#### Contributing

It happens when this *jar*: `org.nd4j.jackson-0.9.1` doesn't contain this `org.nd4j.shade.jackson.module.jaxb.JaxbAnnotationIntrospector` class, and however, Vector transpose in INDArray requires java reflection to this class.

Downgrading the `org.nd4j.jackson` to version *0.7.0* would remove the same exception.

**Comment by agibsonccc** [comment]
Do you have code to reproduce this?

**Comment by cinqs** [comment]
Hi,

We tried to create a basic project to re-produce this issue, but what's wired is that:

when implements this simple snippet:

```java
@PostConstruct
	public void doStart() {
		System.out.println("+++++++++++++++++++++++++++++++++++");
		INDArray a1 = Nd4j.create(new double[][] {{1d, 2d, 3d}, {1d, 2d, 3d}});
		INDArray a = a1.transpose();
	}
```

everything worked fine.

But in our specific project, if use `org.nd4j.jackson-0.9.1`, it happens the first time when we do transpose on a matrix. if we do transpose the second time in a sequence, this exception (in level warning) won't be thrown.

**Comment by raver119** [comment]
That happens only once, because reflection is used only once: to load available compressors for NDArrays.

And 0.7.0 probably works because there were no compression available.

**Comment by cinqs** [comment]
Ok, so? what should we do please?

**Comment by agibsonccc** [comment]
@cinqs could you post a complete pom? Something I'm not understanding here: Why is no one else running in to this? Is this a warning? If this was preventing nd4j from "starting" we would have ran in to this before the release.
Thanks!

**Comment by hongv** [comment]
i meet the same exception,how to resolve ？

**Comment by agibsonccc** [comment]
@hongv so unlike the previous guy, can you help us reproduce this? You are only the second person to run in to this. You are in a very small minority.

**Comment by cinqs** [comment]
sorry for not giving any hint on reproducing the problem. my work previously is on some experimental purpose, and not properly version-controled.

My suggestion on this is to compare `nd4j.jackson`'s version `0.9.1` and `0.7.0`. since I was able to avoid this problem down-grading the package to that version.

again, sorry for not letting you know in time.

**Comment by agibsonccc** [comment]
@cinqs this still doesn't really help us. You're still only 1 of 2 people that I see running in to this.

**Comment by cinqs** [comment]
we can wait to see if @hongv could provide more information.

**Comment by fredo994** [comment]
I managed to reproduce when the error happens. My project is Kotlin project not Java but I'm pretty sure that doesn't change the reason for the error.

When calling `array.getRow(0)` for the first time I get exception:

```
10:09:32.040 [main] WARN  org.reflections.Reflections - could not get type for name org.nd4j.shade.jackson.module.jaxb.JaxbAnnotationIntrospector from any class loader
org.reflections.ReflectionsException: could not get type for name org.nd4j.shade.jackson.module.jaxb.JaxbAnnotationIntrospector
	at org.reflections.ReflectionUtils.forName(ReflectionUtils.java:390) ~[reflections-0.9.11.jar:na]
	at org.reflections.Reflections.expandSuperTypes(Reflections.java:381) [reflections-0.9.11.jar:na]
	at org.reflections.Reflections.<init>(Reflections.java:126) [reflections-0.9.11.jar:na]
	at org.nd4j.linalg.compression.BasicNDArrayCompressor.loadCompressors(BasicNDArrayCompressor.java:41) [nd4j-api-0.9.1.jar:0.9.1]
	at org.nd4j.linalg.compression.BasicNDArrayCompressor.<init>(BasicNDArrayCompressor.java:30) [nd4j-api-0.9.1.jar:0.9.1]
	at org.nd4j.linalg.compression.BasicNDArrayCompressor.<clinit>(BasicNDArrayCompressor.java:23) [nd4j-api-0.9.1.jar:0.9.1]
	at org.nd4j.linalg.factory.Nd4j.getCompressor(Nd4j.java:6306) [nd4j-api-0.9.1.jar:0.9.1]
	at org.nd4j.linalg.api.ndarray.BaseNDArray.get(BaseNDArray.java:4150) [nd4j-api-0.9.1.jar:0.9.1]
	at org.nd4j.linalg.api.ndarray.BaseNDArray.getRow(BaseNDArray.java:4257) [nd4j-api-0.9.1.jar:0.9.1]
        ...
Caused by: java.lang.ClassNotFoundException: org.nd4j.shade.jackson.module.jaxb.JaxbAnnotationIntrospector
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381) ~[na:1.8.0_152]
        ...
```
for class:
```java
public class Main {

	public static void main(String[] args) {
		INDArray array = Nd4j.create(new double[][]{
				{1.0, 2.0, 3.0},
				{2.0, 3.0, 4.0},
				{4.0, 5.0, 6.0}
			}
		);
		INDArray row = array.getRow(0);
	}
}
```

My **pom.xml** file is:
```xml
...
    <properties>
        <nd4j.version>0.9.1</nd4j.version>
        <kotlin.version>1.2.31</kotlin.version>
    </properties>

    <dependencies>
        <dependency>
            <groupId>io.kotlintest</groupId>
            <artifactId>kotlintest-runner-junit5</artifactId>
            <version>3.0.3</version>
            <scope>test</scope>
        </dependency>

        <!-- Logging -->
        <dependency>
            <groupId>ch.qos.logback</groupId>
            <artifactId>logback-classic</artifactId>
            <version>1.0.13</version>
        </dependency>

        <dependency>
            <groupId>org.nd4j</groupId>
            <artifactId>nd4j-native-platform</artifactId>
            <version>${nd4j.version}</version>
        </dependency>

    </dependencies>
...
```
The inclusion of `kotlin-runner-junit5` dependency produced the error.

**Comment by saudet** [comment]
FYI, this should get fixed once we get rid of Reflections, soonish...

**Comment by saudet** [comment]
So, for now, let's consider this as a duplicate of issue #2648.
